app/views/layouts/_sub_sub_menu.html.erb

```diff
@@ -15,43 +15,53 @@
         <% if controller.controller_name != 'assignments' ||
               (controller.controller_name == 'assignments' &&
                controller.action_name != 'new' ) %>
-          <li <% if controller.controller_name == 'assignments' && controller.action_name != 'starter_file' %>
-              class='active'
-              <% end %>>
-            <%= link_to t('menu.properties'), edit_assignment_path(@assignment) %>
-          </li>
-          <li class='<%= "active" if controller.controller_name == 'tags' %>'>
-            <%= link_to Tag.model_name.human.pluralize,
-                        assignment_tags_path(@assignment) %>
-          </li>
-          <li <% if controller.controller_name == 'criteria'%>
-              class='active'
-              <% end %>>
-            <%= link_to Criterion.model_name.human.pluralize,
-                        assignment_criteria_path(@assignment.id) %>
-          </li>
-          <li <% if controller.controller_name == 'assignments' && controller.action_name == 'starter_file' %>
-              class='active'
-              <% end %>>
-            <%= link_to I18n.t('assignments.starter_file.title'),
-                        starter_file_assignment_path(@assignment.id) %>
-          </li>
-          <li class='<%= "active" if controller.controller_name == 'annotation_categories' %>'>
-            <%= link_to Annotation.model_name.human.pluralize,
-                        assignment_annotation_categories_path(@assignment) %>
-          </li>
-          <% if Rails.configuration.x.autotest.enable && !@assignment.is_peer_review? && !@assignment.scanned_exam? %>
-            <li class='<%= "active" if controller.controller_name == 'automated_tests' %>'>
-              <%= link_to t('automated_tests.title'),
-                          manage_assignment_automated_tests_path(@assignment) %>
+          <% if allowed_to?(:manage?, @assignment, with: AssignmentPolicy) %>
+            <li <% if controller.controller_name == 'assignments' && controller.action_name != 'starter_file' %>
+                class='active'
+                <% end %>>
+              <%= link_to t('menu.properties'), edit_assignment_path(@assignment) %>
             </li>
           <% end %>
-          <% if @assignment.scanned_exam? %>
-            <li class='<%= "active" if controller.controller_name == 'exam_templates' %>'>
-              <%= link_to ExamTemplate.model_name.human.pluralize,
-                          assignment_exam_templates_path(@assignment) %>
+          <% if @current_user.admin? %>
```

Comment:
We should try to avoid checking the type of user in views and instead try to check the policies instead.

This applies in a few other places as well

Reply:
Example of "a few other places" that you should look at changing:

- app/views/layouts/menus/_admin_ta_sub_menu_assessments.html.erb